### PR TITLE
Access control enforcement GC and fail on _suite_test.go files missing

### PR DESCRIPTION
**Pull request by GrahamGoudeau**

### CODE REVIEW

File: services/mesh-networking/pkg/access/access-control-enforcer/enforcer_loop.go [hunk 1]
```diff
@@ -66,7 +66,19 @@ func (e *enforcerLoop) Start(ctx context.Context) error {
...
-			logger.Debugf("Ignoring event: %+v", virtualMesh)
+			logger.Debugw("event handler enter",
+				zap.Any("spec", virtualMesh.Spec),
+				zap.Any("status", virtualMesh.Status),
+			)
+
+			// manually set this to false so that things get cleaned up
+			virtualMesh.Spec.EnforceAccessControl = false
+
+			err := e.enforceGlobalAccessControl(ctx, virtualMesh)
```
Comment (harveyxia) [inline services/mesh-networking/pkg/access/access-control-enforcer/enforcer_loop.go:77]: If this fails, we won't get another OnDelete event for the virtualMesh, so the global access control enforcement may be permanently set.

I wonder if this is a valid use case for a requeue

File: ci/assertAllSuitesExist.bash [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#########################################################################
```
Comment (rickducott) [inline ci/assertAllSuitesExist.bash:1]: 👏

File: services/mesh-networking/pkg/access/access-control-enforcer/enforcer_loop.go (hunk 1 above)
Comment (harveyxia) [reply services/mesh-networking/pkg/access/access-control-enforcer/enforcer_loop.go:77]: Discussed offline—we'll leave a comment and make an issue for now, and defer settling on an approach until we have more discussion with the team

Comment (GrahamGoudeau) [reply services/mesh-networking/pkg/access/access-control-enforcer/enforcer_loop.go:77]: https://github.com/solo-io/service-mesh-hub/issues/650

**Pull request merged**